pinot-core/src/main/java/com/linkedin/pinot/core/io/readerwriter/impl/FixedByteSingleColumnSingleValueReaderWriter.java

```diff
@@ -176,8 +187,27 @@ private void addBuffer() {
         new FixedByteSingleValueMultiColWriter(buffer, _numRowsPerChunk, /*cols=*/1, new int[]{_columnSizesInBytes});
 
     final int startRowId = _numRowsPerChunk * (_dataBuffers.size() - 1);
-    _writers.add(new WriterWithOffset(writer, startRowId));
-    _readers.add(new ReaderWithOffset(reader, startRowId));
+    updateHolder(new WriterWithOffset(writer, startRowId), new ReaderWithOffset(reader, startRowId));
+  }
+
+  private void updateHolder(WriterWithOffset newWriter, ReaderWithOffset newReader) {
```

Comment:
Not a blocker for the PR, but would be good to measure and document the overhead of resize.